CONTINUATION.md

```diff
@@ -0,0 +1,511 @@
+`ContinuationFlow`) in each semantic:
+
+```js
+let taskId = 0
+function randomTask() {
+  return ctx.run(`task-${taskId++}`, async () => {
+    await scheduler.wait(Math.random * 10)
+    if (Math.random() >= 0.5) {
+      throw new Error()
+    }
+  })
+}
+```
+
+### `Promise.allSettled`
+
+It is a merge point on `Promise.allSettled`. It aggregates the promise results
+that are either resolved or rejected.
+
+```js
+ctx.run('main', async () => {
+  await Promise.allSettled(_.times(5).map((it) => randomTask()))
+  ctx.get() // => (1)
+})
+```
+
+From different observation perspectives, the value at site (1) would be
+different on the above semantics:
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, the proposed solution didn't specify
+  conflicts-resolving yet. If the default behavior merges the values in an
+  array, it changes the shape of the value.
+
+### `Promise.all`
+
+In the resolving path, `Promise.all` is similar to `Promise.allSettled` that it
+is a merge point when all promises resolve.
+
+`Promise.all` picks the fastest reject promise.
+
+```js
+ctx.run('main', async () => {
+  try {
+    await Promise.all(_.times(5).map((it) => randomTask()))
+    ctx.get() // => (1)
+  } catch (e) {
+    ctx.get() // => (2)
+  }
+})
+```
+
+The value at site (1):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, the proposed solution didn't specify
+  conflicts-resolving yet.
+
+The value at site (2):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, it is the one caused the rejection, and discarding
+  leaf contexts of promises that may have been fulfilled.
+
+### `Promise.race`
+
+`Promise.race` picks the fastest settled (either fulfilled or rejected) promise.
+
+```js
+ctx.run('main', async () => {
+  try {
+    await Promise.race(_.times(5).map((it) => randomTask()))
+    ctx.get() // => (1)
+  } catch (e) {
+    ctx.get() // => (2)
+  }
+})
+```
+
+The value at sites (1) and (2):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, it is the one caused the settlement.
+
+### `Promise.any`
+
+In a happy path, `Promise.any` is similar to `Promise.race` that it picks the
+fastest resolved promise. However, it aggregates the exception values when all
+the promises are rejected.
+
+It is a merge point when exception values are aggregated.
+
+```js
+ctx.run('main', async () => {
+  try {
+    await Promise.any(_.times(5).map((it) => randomTask()))
+    ctx.get() // => (1)
+  } catch (e) {
+    ctx.get() // => (2)
+  }
+})
+```
+
+The value at site (1):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, it is the one caused the fulfillment, and discarding
+  leaf contexts of promises that may have been rejected.
+
+The value at site (2):
+
+- For `AsyncContext.Variable`, it is `main`,
+- For `ContinuationFlow`, the proposed solution didn't specify
+  conflicts-resolving yet.
```

Comment:
I would do the same `AggregateContext` thing I mentioned above. It may also be helpful to have some sort of positional helper to map contexts from the set to rejections from the original promise set though. This one is also a bit more complicated though as other contexts may not even be _complete_ yet when a rejection terminates the listening to of the _other_ promises. The `AggregateContext` type may also need a `pending` state per slot to match the promises those slots respresent.